hazelcast/src/main/java/com/hazelcast/internal/ascii/memcache/IncrementCommandProcessor.java

```diff
@@ -68,31 +71,23 @@ public void handle(IncrementCommand incrementCommand) {
 
     private void incrementUnderLock(IncrementCommand incrementCommand, String key, String mapName) {
         Object value = textCommandService.get(mapName, key);
-        MemcacheEntry entry;
 
         if (value != null) {
-            if (value instanceof MemcacheEntry) {
-                entry = (MemcacheEntry) value;
-            } else if (value instanceof byte[]) {
-                entry = new MemcacheEntry(incrementCommand.getKey(), (byte[]) value, 0);
-            } else if (value instanceof String) {
-                entry = new MemcacheEntry(incrementCommand.getKey(), stringToBytes((String) value), 0);
-            } else {
-                try {
-                    entry = new MemcacheEntry(incrementCommand.getKey(), textCommandService.toByteArray(value), 0);
-                } catch (Exception e) {
-                    throw ExceptionUtil.rethrow(e);
-                }
-            }
+            MemcacheEntry entry = entryConverter.toEntry(incrementCommand.getKey(), value);
 
-            final byte[] value1 = entry.getValue();
-            final long current = (value1 == null || value1.length == 0) ? 0 : byteArrayToLong(value1);
-            long result = -1;
-            result = incrementCommandTypeCheck(incrementCommand, result, current);
+            String currentCachedValue = bytesToString(entry.getValue());
```

Comment:
If value1 (`entry.getValue()`) is of size 0, then `currentCachedValue` will be an empty string. And `Long.parseUnsignedLong("")` will throw a `NumberFormatException`. This is no different than trying to increment a value "foo", for example. I added a new test `testIncrement_onEmptyValue_shouldFail` for this case, just to be safe though. Note that the previous implementation treated empty strings as 0, and that was incorrect.

About `value1 == null`, I don't think it's possible. `entry` comes either from an IMap or constructed in `EntryConverter.toEntry`. If it came from an IMap, then it is deserialized using its `readData` method, which sets the `value` variable to a nonnull value. If it was constructed in `toEntry`, then again, it can't be null.